### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by SaboorSohaib**
### Hi Dear!

**We implemented the following requirements.**

- Describe the SOLID principles of OOP.
- Refactor your own Ruby code after gaining new knowledge.
- Apply Ruby best practices and language style guides in code.

**Thank you.**

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,41 +1,5 @@
-#!/usr/bin/env ruby
-require_relative 'app'
-# rubocop:disable Metrics/CyclomaticComplexity
-def main
-  app = App.new
+require_relative 'starting'
 
-  response = nil
+def main; end
```
Comment (Gambit142) [inline main.rb]: - [ ] Currently, your main method does nothing. The code that starts the application is in the `starting.rb` file and it originates from the Start class. (proof is when I run `ruby starting.rb` in my terminal, the application starts). Based on the requirements, the entry point for the application should be in the `main.rb` file in a main method. Kindly implement this requirement. 👍

File: nameable.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Nameable
   def correct_name
-    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+    raise NotImplementedError
   end
 end
```
Comment (Gambit142) [inline nameable.rb:5]: I can't highlight the relevant codes to buttress my point, but I will explain it here.

 - [ ] The essence of the Single Responsibility Rule is to ensure that functions/classes are responsible for a single job or thing. For example, in this application, the method for the user's input and input validation should be a different method from the method that creates the person or book. I would advise that you put this in a different method and call the method here. Kindly refer to this [documentation](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) 👍 💻 

In your `App` class, the following methods should have the code for taking user input different from the ones that create the object (instance of the class):
- `create_a_perspon`
- `create_student`
- `create_teacher`
- `create_a_rental`
- `create_a_book`

#### For example, in the `create_a_book` method, the set of codes in the screenshot below should be put in a separate method or module:

image

#### While these should be put in a separate method or module

image


#### Hint: I would advise you to create helper modules for each scenario. 👍

File: starting.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    response = gets.chomp
+
+    case response
+    when '1'
+      app.list_all_books
+    when '2'
+      app.list_all_people
+    when '3'
+      app.create_a_person
+    when '4'
```
Comment (Gambit142) [inline starting.rb]: - [ ] In the application, when you are creating a Teacher, the input shouldn't have permissions based on the video requirements. Kindly change this

https://www.youtube.com/watch?v=vkkgrhD6aXQ

#### I would advise you to follow the video to know what should be printed out for the user. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib @Ngufor-emmanuella 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- README.md file is professional ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: nameable.rb (hunk 2 above)
Comment (SaboorSohaib) [reply nameable.rb:5]: Hi @Gambit142 
Thank you for reviewing our project. Even though I studied the provided documentation many times and did not get the whole idea I  will be appreciated it if you refer us to some other tutorials or documentation.

**Comment by Gambit142** [comment]
Hi @SaboorSohaib, if you are having a difficult time understanding the material provided, you can check out these articles to have a better understanding:

- https://www.honeybadger.io/blog/ruby-solid-design-principles/
- https://subvisual.com/blog/posts/19-solid-principles-in-ruby/
- https://rubygarage.org/blog/solid-principles-of-ood

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @SaboorSohaib, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**